internal/services/containers/kubernetes_fleet_manager_resource.go

```diff
@@ -183,45 +237,39 @@ func (r KubernetesFleetManagerResource) Update() sdk.ResourceFunc {
 				return err
 			}
 
-			var config KubernetesFleetManagerResourceSchema
-			if err := metadata.Decode(&config); err != nil {
-				return fmt.Errorf("decoding: %+v", err)
-			}
-
-			existing, err := client.Get(ctx, *id)
-			if err != nil {
-				return fmt.Errorf("retrieving existing %s: %+v", *id, err)
-			}
-			if existing.Model == nil {
-				return fmt.Errorf("retrieving existing %s: properties was nil", *id)
-			}
-			payload := *existing.Model
-
-			r.mapKubernetesFleetManagerResourceSchemaToFleet(config, &payload)
-
-			if err := client.CreateOrUpdateThenPoll(ctx, *id, payload, fleets.DefaultCreateOrUpdateOperationOptions()); err != nil {
-				return fmt.Errorf("updating %s: %+v", *id, err)
+			if err := client.DeleteThenPoll(ctx, *id, fleets.DefaultDeleteOperationOptions()); err != nil {
+				return fmt.Errorf("deleting %s: %+v", id, err)
 			}
 
 			return nil
 		},
 	}
 }
 
-func (r KubernetesFleetManagerResource) mapKubernetesFleetManagerResourceSchemaToFleet(input KubernetesFleetManagerResourceSchema, output *fleets.Fleet) {
-	output.Location = location.Normalize(input.Location)
-	output.Tags = tags.Expand(input.Tags)
+func expandFleetHubProfileModel(inputList []FleetHubProfileModel) *fleets.FleetHubProfile {
+	if len(inputList) == 0 {
+		return nil
+	}
 
-	if output.Properties == nil {
-		output.Properties = &fleets.FleetProperties{}
+	input := inputList[0]
+	output := &fleets.FleetHubProfile{
+		DnsPrefix: pointer.To(input.DnsPrefix),
 	}
+
+	return output
 }
 
-func (r KubernetesFleetManagerResource) mapFleetToKubernetesFleetManagerResourceSchema(input fleets.Fleet, output *KubernetesFleetManagerResourceSchema) {
-	output.Location = location.Normalize(input.Location)
-	output.Tags = tags.Flatten(input.Tags)
+func flattenFleetHubProfileModel(input *fleets.FleetHubProfile) []FleetHubProfileModel {
+	if input == nil {
+		return nil
```

Comment:
We should return an empty hub profile model here
```suggestion
		return []FleetHubProfileModel
```